### PR TITLE
Adding default to anonymous column

### LINKED ISSUE #5601: Upgrade to 2.2.1 with PGSQL fails:  column "anonymous" contains null values
| Q | A
| --- | ---
| Sulu Version | 2.2.1

# Actual Behavior

```
doctrine:schema:update --force   

 Updating database schema...


In AbstractPostgreSQLDriver.php line 51:
                                                                                                
  An exception occurred while executing 'ALTER TABLE se_roles ADD anonymous BOOLEAN NOT NULL':  
                                                                                                
  SQLSTATE[23502]: Not null violation: 7 ERROR:  column "anonymous" contains null values        
                                                                                                

In PDOConnection.php line 91:
                                                                                          
  SQLSTATE[23502]: Not null violation: 7 ERROR:  column "anonymous" contains null values  
                                                                                          

In PDOConnection.php line 86:
                                                                                          
  SQLSTATE[23502]: Not null violation: 7 ERROR:  column "anonymous" contains null values  
                                                                                          

```

# Expected Behavior

Schema can be updated

# Steps to Reproduce

Run update with pgsql

# Possible Solutions

I guess a default has to be set on a NULL column or pgsql will fail. Will see if I can make a pull request tomorrow.

### PR DESCRIPTION
**Pull request by kfrohwein**
Fixes #5601

| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #5601
| License | MIT

#### What's in this PR?

Added a default for the new anonymous column

#### Why?

PGSql won't be able to upgrade

### CODE REVIEW

File: src/Sulu/Bundle/SecurityBundle/Resources/config/doctrine/Role.orm.xml [hunk 1]
```diff
@@ -11,7 +11,11 @@
         <field name="name" type="string" column="name" length="60" unique="true"/>
         <field name="key" type="string" length="60" column="role_key" nullable="true" unique="true"/>
         <field name="system" type="string" column="securitySystem" length="60"/>
-        <field name="anonymous" type="boolean" column="anonymous" nullable="false"/>
+        <field name="anonymous" type="boolean" column="anonymous" nullable="false">
+            <options>
+                <option name="default">false</option>
```
Comment (nnatter) [inline src/Sulu/Bundle/SecurityBundle/Resources/config/doctrine/Role.orm.xml:16]: Looks like we need to use the following here:
```suggestion
                <option name="default">0</option>
```

Can you test if this works on PGSQL?

**Comment by alexander-schranz** [comment]
Thank you for this PR we should also add a line to the `UPGRADE.md` with the ALTER statement so the database change is also documented there:

```sql
ALTER TABLE `se_roles` CHANGE `anonymous` `anonymous` TINYINT(1) NOT NULL DEFAULT 0;
```

See https://github.com/sulu/sulu/blob/master/UPGRADE.md#role-entity-changed-for-anonymous-roles es example.

**Comment by nnatter** [comment]
Updated this PR for you 😇

**Comment by nnatter** [comment]
@kfrohwein Can you test if this solves the problem for you?

**Comment by alexander-schranz** [comment]
Would be nice if we could merge this to close the issue with postgresql.

/cc @kfrohwein can you confirm the changes by @nnatter works for your postgresql version?

**Comment by nnatter** [comment]
Tested this PR using Postgres 12.5 with the following steps:
1. Checkout sulu `release/2.1`
2. `bin/console doctrine:schema:update --force`
3. Checkout sulu `release/2.2`
4. `bin/console doctrine:schema:update --force`

```
 Updating database schema...


In AbstractPostgreSQLDriver.php line 64:

  An exception occurred while executing 'ALTER TABLE se_roles ADD anonymous BOOLEAN NOT NULL':

  SQLSTATE[23502]: Not null violation: 7 ERROR:  column "anonymous" contains null values


In Exception.php line 18:

  SQLSTATE[23502]: Not null violation: 7 ERROR:  column "anonymous" contains null values


In PDOConnection.php line 125:

  SQLSTATE[23502]: Not null violation: 7 ERROR:  column "anonymous" contains null values

```

5. Checkout sulu `release/2.1`
6. `bin/console doctrine:schema:update --force`
7. Checkout sulu `kfrohwein:patch-4`
8. `bin/console doctrine:schema:update --force`

```
 Updating database schema...

     1 query was executed


 [OK] Database schema updated successfully!
```

So I think this looks good! 👍

**Review by nnatter — APPROVED** [review]

**Pull request merged**